### PR TITLE
OW-105: New setMetadata top-level function

### PR DESCRIPTION
**Pull request by joliveraortega**

### CODE REVIEW

**Comment by joliveraortega** [comment]
Some telemetry payload being sent because of the new function added. Now fields `"applicationId":"vonage-media-processor-example"` and `"source":"test"` are sent.

```
{"action":"MediaProcessor","applicationId":"vonage-media-processor-example","timestamp":1645106050449,"guid":"f2765b49-3339-4f15-bdb1-e3f51a858d11","source":"test","variation":"Create"}
{"action":"MediaTransformer","applicationId":"vonage-media-processor-example","timestamp":1645106050453,"guid":"0a4a0e62-595b-4cff-b4d8-78204993b91f","source":"test","transformerType":"Custom","variation":"Create"}
```

**Comment by joliveraortega** [comment]
@gmininberg This one is ready for review too. This PR depends on https://github.com/Vonage/media-processor/pull/2 so it has to be reviewed and merged in after.

File: lib/src/telemetry/Reporter.ts [hunk 1]
```diff
@@ -2,39 +2,60 @@ import * as pack from './../../../package.json';
...
+export type VonageSourceType = 'automation' | 'test' | 'vbc' | 'video' | 'voice';
+
+export type VonageMetadata = {
+  sourceType: VonageSourceType,
+  appId: string
+}
+
+let _metadata: VonageMetadata;
+
+export function setMetadata(metadata: VonageMetadata): Promise<void> {
```
Comment (gmininberg) [inline lib/src/telemetry/Reporter.ts:14]: in my opinion no need promise here since its just set function and not really do something.

**Comment by joliveraortega** [comment]
@gmininberg Review comments addressed. Also bumped up version.

File: lib/src/telemetry/Reporter.ts (hunk 1 above)
Comment (joliveraortega) [reply lib/src/telemetry/Reporter.ts]: Addressed.

**Review by gmininberg — APPROVED** [review]

**Pull request merged**